rse-r-package.Rmd

```diff
@@ -0,0 +1,590 @@
+    CITATION.md
+  Most likely ‘inst/CITATION’ should be used instead.
+
+0 errors ✔ | 1 warning ✖ | 3 notes ✖
+Error: R CMD check found WARNINGs
+Execution halted
+
+Exited with status 1.
+```
+
+The explanations for these are:
+
+1.  We haven't written any documentation for the `count_words` function.
+2.  We haven't described our license in the approved way.
+3.  Three of the files we added for GitHub's benefit have to be renamed or moved.
+
+After a bit of searching online,
+we move `LICENSE.md` to `LICENSE` with no extension (but still in the project's root directory)
+and change the entry in `DESCRIPTION` for the license to `License: MIT + file LICENSE` (spelled exactly that way).
+We also add lines to `.Rbuildignore` to tell RStudio to ignore `CITATION.md` and `CONDUCT.md`:
+
+```{r, child="rse-r-package/zipffreq/.Rbuildignore"}
+```
+
+We could change the name of `CITATION.md` to `inst/CITATION` instead,
+so that it will be copied into the root of the installation directory on users' machines,
+bit we decide to give priority to GitHub's convention that
+the citation description in the project's root directory.
+
+### Exercise: Packaging the CO2 functions
+
+1.  Create a brand-new package to hold the CO2 data and functions for analyzing it.
+2.  Put the license and citation files where R's packaging rules want them.
+3.  Build the package: what warnings do you get and what do they mean?
+
+### Exercise: Exploring a package
+
+1.  Clone a GitHub repository that contains an R package, such as <https://github.com/r-lib/usethis>.
+2.  Which files in this repository do you recognize, and what are they for?
+3.  Which files are there to satisfy something other than R's packaging system?
+
+## How do I document a R package? {#rse-r-package-document}
+
+Our next task is to document our function.
+To do this,
+we turn to Hadley Wickham's *[R Packages][wickham-packages]*
+and Karl Broman's "[R package primer][broman-packages]"
+for advice on writing roxygen2.
+We then return to our source file and put a specially-formatted comment in front of our code:
+
+```{r, child="rse-r-package/etc/frequency-doc.R"}
+```
+
+This comment is an example of [embedded documentation](glossary.html#embedded-documentation).
+Over the years,
+programmers have found that if they put code in one file and documentation in another,
+the documentation quickly falls out of date with the code:
+people will change a function's name or add a new parameter and forget to update the docs.
+If the documentation lives right beside the code,
+on the other hand,
+the next person to modify the code is far more likely to remember to update it.
+Tools like roxygen2 can read the code,
+extract the documentation,
+and format it as HTML or PDF.
+They can also do things like create an index,
+which would be even more painful to do by hand than writing the documentation itself.
+
+roxygen2 processes comment lines that start with `#'` (hash followed by single quote).
```

Comment:
I suggest running `usethis::use_roxygen_md()` to allow the docs to be written in Markdown

Reply:
I haven't done this - can I ask you to rewrite this section to demonstrate?

Reply:
@gvwilson sure, no problem.